hax/hax/handler.py

```diff
@@ -138,6 +155,16 @@ def update_process_failure(self, q: Queue,
                         chp_type=m0HaProcessType.M0_CONF_HA_PROCESS_M0D,
                         chp_pid=0,
                         fid=state.fid))
+                # We need to check whether the process is failed due to node
+                # failure. If there is a node failure, hax confirms it from
+                # consul and updates the list for failed motr configuration
+                # objects.
+                node = self.consul.get_process_node(state.fid)
+                node_status = self.consul.get_node_health(node)
+                LOG.debug('node: %s status: %s', node, node_status)
+                if (node_status != 'passing'):
+                    self.update_node_failure(
```

Comment:
@Shreya-18, you need to call you function in hax/hax/motr/__init__.py::broadcast_ha_states() function,
```
    def broadcast_ha_states(self, ha_states: List[HAState],
                            notify_devices=True) -> List[MessageId]:
        LOG.debug('Broadcasting HA states %s over ha_link', ha_states)

        def ha_obj_state(st):
            return HaNoteStruct.M0_NC_ONLINE if st.status == ServiceHealth.OK \
                else HaNoteStruct.M0_NC_FAILED

        notes = []
        for st in ha_states:
            if st.status in (ServiceHealth.UNKNOWN, ServiceHealth.OFFLINE):
                continue
            note = HaNoteStruct(st.fid.to_c(), ha_obj_state(st))
            notes.append(note)
            if (st.fid.container == ObjT.PROCESS.value and
                    st.status == ServiceHealth.STOPPED):
                notify_devices = False
            notes += self._generate_sub_services(note, self.consul_util,
                                                 notify_devices)
```
After generate_sub_services() you can call your function that returns a list of HAState. The list must be appended to the `notes`. 
So if its a process failure then we will report failure for the corresponding node (enclosure) and CVGs.

Reply:
Thanks, have updated the patch and tested it. Works fine!!